### ISSUE TITLE
A11y Fail: Table + Data grid - Need to use Caption as title of table

### ISSUE THREAD

**Issue by orion-cengage** [cengage/react-magma#1395]
CENG_1953

https://react-magma.cengage.com/version/4.5.0/api/datagrid/

https://react-magma.cengage.com/version/4.5.0/api/table/

### Issue:
It is a best practice to ensure that all <table> elements have a <caption> element. A caption functions like a heading for a table. Most screen readers announce the content of captions. Captions help users to find a table and understand what it’s about and decide if they want to read it. If the user uses “Tables Mode”, captions are the primary mechanism to identify tables.

### Suggestion:
We need to document and make it easy for devs to use "caption" attribute to title tables and data grids. Additionally, we need to let them know if they want the caption to also be a heading within the structure of the page, they can do both at the same time.

`<table><caption><h2>This is the title of the table</h2></caption>...`

**Comment by silvalaura** [comment]
We should add a new prop to Table and DataGrid to accept a heading node. We should provide default styling, but we should support any heading type.

**Comment by orion-cengage** [comment]
I have illustrated in Figma how the caption should appear both when the table has borders turned on and off.

[Figma link](https://www.figma.com/design/z2H2qv9iYQX942LqToCuQE/Components---Data-Display?node-id=2319-14877&t=yJKCVkiukMgyQ869-1)

While we're in there, lets fix once and for all the inconsistencies we have in the rounded corners. And let's keep it simple: 

- No more rounded corners on tables. Please remove them.

![Image](https://github.com/user-attachments/assets/00159a3b-bdf7-48da-af66-ed89f936660d)

- This goes for both Table and Datagrid
- Verify if you are using the Pagination that attaches to the bottom, you also don't have rounded corners

![Image](https://github.com/user-attachments/assets/a44788ba-621d-4718-a460-f77a52c62b71)

- Verify if you are using the Pagination that attaches to the bottom, it also gets appropriate borders

![Image](https://github.com/user-attachments/assets/5ca40c96-b2bf-4334-8625-d83dbfb39bf4)

**Comment by olehnoskov-cengage** [comment]
Add a new prop or possibility easily adding padding for the table.
